cognite/powerops/resync/v1/utils.py

```diff
@@ -0,0 +1,307 @@
+# Mypy does not understand the pydantic classes that allows both alias and name to be used in population
+# https://github.com/pydantic/pydantic/issues/3923
+# mypy: disable-error-code="call-arg"
+from __future__ import annotations
+
+import logging
+import random
+import re
+from typing import Any, Optional
+
+from pydantic.alias_generators import to_snake
+
+import cognite.powerops.client._generated.v1.data_classes as v1_data_classes
+
+logger = logging.getLogger(__name__)
+
+
+def get_prefix_from_type(domain_model_type: type) -> str:
+    """Get the type prefix for a domain model type.
+
+    Given the domain model type, this function returns the type prefix as a snake case string. Also strips the "Write"
+    suffix from the type name to get the root type name.
+
+    Args:
+        domain_model_type: The domain model type to get the prefix for.
+
+    Returns:
+        The type prefix as a snake case string.
+    """
+
+    if issubclass(domain_model_type, v1_data_classes.DomainModelWrite):
+        type_name = domain_model_type.__name__.removesuffix("Write")
+    else:
+        # TODO: should check if it's in the non write classes
+        type_name = domain_model_type.__name__
+        logger.warning(f"Type {type_name} is not a subclass of DomainModelWrite")
+
+    # Convert the type name to snake case
+    external_id_prefix = to_snake(type_name).lower()
```

Comment:
to_snake already does .lower(), so the .lower() call here is redundant
```suggestion
    external_id_prefix = to_snake(type_name)
```